lua/entities/gmod_wire_expression2/core/entity.lua

```diff
@@ -392,8 +392,15 @@ e2function number entity:getSkin()
 end
 
 --- Sets <this>'s skin number.
-e2function void entity:setSkin(skin)
-	if IsValid(this) then this:SetSkin(skin) end
+e2function void entity:setSkin(skinIndex)
+	-- Our entity must be a valid non-player entity, and it has to be owned by E2's player
+	if IsValid(this) and not this:IsPlayer() and isOwner(self, this)
```

Comment:
This shouldn't check `isOwner`; the `CanProperty` check should be all that's needed. That way, prop protection can allow friends to change skins of each other's models, etc - the point of calling a hook is that we shouldn't make any prop protection policy decisions ourselves.